### PR TITLE
add pr checks

### PR DESCRIPTION
**Pull request by pkucmus**

### CODE REVIEW

File: .github/workflows/check-pr.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install Python dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-root
+
+      - name: Run linters
+        run: |
+          poetry run black src/
```
Comment (korycins) [inline .github/workflows/check-pr.yml:41]: IMO, all these commands could be called on sample dir also

**Comment by korycins** [comment]
Also, I would limit it to a single newest python version.

**Comment by pkucmus** [comment]
> Also, I would limit it to a single newest python version.

I already need the framework to work on both 3.9 and 3.10 so I'd keep the ones the framework is intended to work on

File: .github/workflows/check-pr.yml (hunk 1 above)
Comment (pkucmus) [reply .github/workflows/check-pr.yml]: Good call, those were not formated at all

**Review by korycins — APPROVED** [review]

**Pull request merged**